test/modules/acbs/acbs-authentication.service.test.ts

```diff
@@ -0,0 +1,242 @@
+jest.mock('@nestjs/axios');
+import { HttpService } from '@nestjs/axios';
+import { AxiosError, AxiosHeaders } from 'axios';
+import { of, throwError } from 'rxjs';
+import { when } from 'jest-when';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+import { AcbsAuthenticationFailedException } from '@ukef/modules/acbs/acbs-authentication-failed.exception';
+
+describe('AcbsAuthenticationService', () => {
+    const baseUrl = 'the base url';
+    const loginName = 'the login name';
+    const password = 'the password';
```

Comment:
Can we install `chance.js` as a dev dependency to randomise these values? It should stop sonarcloud from flagging it as an issue as well

Reply:
I usually prefer not to use randomised test data in unit tests because:
1. it means the test inputs aren't repeatable,
2. if you get an error mentioning the value of a particular variable, you can't tell _which_ variable is being mentioned

I've made the change anyway and set a constant seed for `chance.js` so that hopefully takes care of issue (1).

I think issue (2) is still a minor downside but I'm happy to keep it this way if it's more consistent across the other projects - let me know what you think

Reply:
Usually yes, but the thinking here is to test it with scenarios that we didn't consider. 
I.e. a random string could generate a scenario that we didn't consider before and could break the functionality. 
It won't be applicable for every test, but it's good to balance it out